apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1205,27 +1205,34 @@ handle_async_worker_down(Data0, Pid) ->
 -spec call_query(force_sync | async_if_possible, _, _, _, _, _) -> _.
 call_query(QM, Id, Index, Ref, Query, QueryOpts) ->
     ?tp(call_query_enter, #{id => Id, query => Query, query_mode => QM}),
-    case emqx_resource_manager:lookup_cached(extract_connector_id(Id)) of
+    case emqx_resource_cache:get_runtime(Id) of
```

Comment:
Since `emqx_resource_cache:get_runtime/1` apparently expects only channel IDs, won't this prevent calling the connector with special queries, like calling postgres with `{query, "select * from ..."}`?